plugins/modules/ec2_vol.py

```diff
@@ -809,6 +815,14 @@ def main():
                         changed=False
                     )
 
+        if module.check_mode:
```

Comment:
@mandar242  Thank you for working on this. What would be the behaviour if creating or modifying a volume in check mode? At this stage, I don't think it is handled. The check mode should cover all the module's functionality. 

I think this code here could also be removed. It will be better to manage the check mode at function level wether possible.

Reply:
Hi @alinabuzachis, Thank you for feedback.
From local testing, I believe this code block handles cases for create/update/detach volume in check_mode. 
Please let me know if you suggest modifying it to more function level, although I think the output would remain same in either case.
[818-824](https://github.com/mandar242/amazon.aws/blob/ec2_vol_check_mode_support/plugins/modules/ec2_vol.py#L818)

Reply:
@mandar242 what I was suggesting is instead of stopping here, try to follow the flow and manage the check mode inside create_volume() or update_volume() for example. Similar to what you did in the PR where you implemented the check mode for ec2_snapshot. I also suggest adding some other tests to cover the other functionalities of the module using check mode ( modify a volume, idempotence, etc)